src/drivers/ins/eulernav_bahrs/eulernav_driver.h

```diff
@@ -0,0 +1,177 @@
+	};
+
+	/// @brief An object for grouping message-related attributes
+	class NextMessageInfo
+	{
+	public:
+		bool _is_detected{false}; ///< A flag to indicate that the next message header was detected
+		uint16_t _protocol_version{0U}; ///< Protocol version retrieved from the next message header
+		uint8_t _message_code{0U}; ///< Message code retrieved from the next message header
+	};
+
+	/// @brief Perform initialization
+	/// If the driver is not initialized, the function does the following:
+	/// 1. Opens serial port
+	/// 2. Allocates a ring buffer for RX data stream
+	/// 3. Cleans up if any of the previous operations fails
+	/// 4. Sets the "is initialized" flag to true on success
+	void initialize();
+
+	/// @brief De-initialize
+	/// The function does the following:
+	/// 1. Close the serial port, if it is opened
+	/// 2. Deallocates the ring buffer
+	/// 3. Resets the "is initialized" flag to false
+	void deinitialize();
+
+	/// @brief Process data in the ring buffer.
+	/// Extracts and parses protocol messages.
+	void processDataBuffer();
+
+	/// @brief Searches the ring buffer for sync bytes.
+	/// @param buffer Ring buffer to search
+	/// @return True if sync bytes found, false if the number of bytes remaining in the buffer is less then message header length.
+	static bool findNextMessageHeader(Ringbuffer &buffer);
+
+	/// @brief Get protocol version and message code from the ring buffer.
+	/// @param buffer The buffer to process
+	/// @param protocol_ver Output protocol version
+	/// @param message_code Output message code
+	/// @return True on success, false on failure
+	static bool retrieveProtocolVersionAndMessageType(Ringbuffer &buffer, uint16_t &protocol_ver, uint8_t &message_code);
+
+	/// @brief Decode a message from BAHRS and publish its content.
+	/// @param data An array of message bytes
+	/// @param messsage_id Message ID
+	void decodeMessageAndPublishData(const uint8_t *data, CSerialProtocol::EMessageIds messsage_id);
+
+	/// @brief Decode and publish IMU data.
+	/// @param data Inertial message data bytes
+	void handleInertialDataMessage(const uint8_t *data);
+
+	/// @brief Decode and publish vehicle attitude and pressure height.
+	/// @param data Navigation message data bytes
+	void handleNavigationDataMessage(const uint8_t *data);
+
+	/// @brief Get message length by message ID
+	/// @param messsage_id Query message ID
+	/// @return Message length, or -1 if the message ID is unknown or not supported.
+	static int32_t getMessageLength(CSerialProtocol::EMessageIds messsage_id);
+
+	/// @brief Perform CRC
+	/// @param buf Data buffer pointer
+	/// @param len Number of words to include in the CRC.
+	/// @return CRC value
+	static uint32_t crc32(const uint32_t *buf, size_t len);
+
+	using VehicleAttitude = px4::msg::VehicleAttitude;
+	using PressureData = px4::msg::SensorBaro;
```

Comment:
the `using` here isn't necessary, both of these are only used twice. I would prefer to have the verbose name `px4::msg::VehicleAttitude` and `px4::msg::SensorBaro` so that it is clear what these variables are (uorb messages) when reading through the code